### DISCUSSION TITLE
Allow self-signed provider

### DISCUSSION THREAD

**Discussion by Nautigsam** [rayluo/identity#30] [Q&A] [answered]
I want to be able to use this library with a local IdP for testing purpose (like Keycloak). I want to use self-signed certificates so I need to disable TLS verifications in that case.

MSAL already offers that as a parameter for `ConfidentialApplication`, so we just need to mirror it on `Auth` class.

**Comment by rayluo** [comment]
If you are referring to the `oidc_authority` parameter in MSAL, it is already available in `identity`'s `Auth` class, as documented in the [purple box here](https://identity-library.readthedocs.io/en/latest/django.html#configuration).

Please let me know by either clarifying the parameter or a thumbs-up and/or closing this issue.

**Comment by Nautigsam** [comment]
I am sorry I wasn't more specific: I was referring to the [verify argument](https://msal-python.readthedocs.io/en/latest/#msal.ClientApplication.params.verify).

**Comment by rayluo** [comment]
I haven't tried it but, assuming you are also using the `requests` library for http traffic, perhaps you can give [this workaround](https://stackoverflow.com/questions/15445981/how-do-i-disable-the-security-certificate-check-in-python-requests#comment104417247_58955034) a try.

    export REQUESTS_CA_BUNDLE='your-ca.pem'

**Comment by Nautigsam** [comment]
It does not seem to work with self-signed certificate 😕

**Comment by rayluo** [comment]
Hmm, there were some slightly different workarounds mentioned that Q&A. What I meant was this one, `export REQUESTS_CA_BUNDLE='your-ca.pem'`, which some other commenters reported success with. Can you elaborate what you have tried that "does not seem to work"?

**Comment by Nautigsam** [comment]
I did set `REQUESTS_CA_BUNDLE` with the path to my Keycloak self-signed certificate. It had no effect.

**Comment by rayluo** [comment, marked as answer]
> I did set `REQUESTS_CA_BUNDLE` with the path to my Keycloak self-signed certificate. It had no effect.

I don't know what is inside your "keycloak self-signed certificate". Given that the `REQUESTS_CA_BUNDLE` approach is a documented feature of `requests`, it is supposed to work. Perhaps, you can double check by using these detailed steps of [how to compose that ca bundle file](https://stackoverflow.com/a/48636689/728675). Someone even came up with [a script for that process](https://stackoverflow.com/a/77270973/728675).

**Comment by Nautigsam** [comment]
My certificate was faulty, I missed `localhost` as an Subject Alternative Name.

I am sorry for the false alert, `REQUESTS_CA_BUNDLE` is definitely the way to go. Thank you for your help.
